CONTRIBUTING.md

```diff
@@ -45,22 +45,132 @@ To add a new plugin you need to do the following.
+thanks to package referring to the `name` attribute.
+
+#### `mkVimPlugin`
+
+The `mkVimPlugin` function provides a standardized way to create a `Vim` plugin.
+This is intended to be used with traditional vim plugins, usually written in viml.
+Such plugins are usually configured via vim globals, but often have no configurable options at all.
+
+| Parameter                | Description                                                                 | Required | Default Value |
+|--------------------------|-----------------------------------------------------------------------------|----------|---------------|
+| **name**                 | The name of the Vim plugin.                                                 | Yes      | N/A           |
+| **url**                  | The URL of the plugin repository.                                           | Yes      | N/A           |
+| **colorscheme**          | The name of the colorscheme.                                                | No       | `name` parameter |
+| **deprecateExtraConfig** | Flag to deprecate extra configuration.                                      | No       | `false`       |
+| **description**          | A description of the plugin. Can also be used for non-normative documentation, warnings, tips and tricks.                                                | No       | `null`        |
+| **extraConfig**          | Extra configuration for the plugin.                                         | No       | `cfg: {}`     |
+| **extraOptions**         | Extra options for the plugin.                                               | No       | `{}`          |
+| **extraPackages**        | Extra packages to include.                                                  | No       | `[]`          |
+| **extraPlugins**         | Extra plugins to include.                                                   | No       | `[]`          |
+| **globalPrefix**         | Global prefix for the settings.                                             | No       | `""`          |
+| **imports**              | A list of imports for the plugin.                                           | No       | `[]`          |
+| **isColorscheme**        | Flag to indicate if the plugin is a colorscheme.                            | No       | `false`       |
+| **maintainers**          | The maintainers of the plugin.                                              | No      | `[]` |
+| **optionsRenamedToSettings** | List of options renamed to settings.                                    | No       | `[]`          |
+| **originalName**         | The original name of the plugin, usually the plugin's github repo name. E.g. `"foo-bar.nvim"`.                                            | No       | `name` parameter |
+| **package**              | The package for the plugin.                                                 | No       | `name` parameter |
```

Comment:
Minor nit: this is an intrinsic issue with markdown tables, but the column deliminators have started creeping out of alignment... Same for `mkNeovimPlugin`.

Is your goal to set each column divider at a particular char-column and then just have a few exceptions for rows with particularly wordy fields?

Maybe just have a final check when all else is done that the table is formatted/aligned as you intended :grin:

Reply:
Hey! TBH, I really hate MD tables due to the lack of readability on large columns... So I tried my best in my latest change to fit a *readable* design. Tell me what you think. Another shi**y solution would be to create an over-engineered method with a json/yaml input, a simple CLI command to template this and `sed` the content. So, you don't interact with raw MD tables and just the read the rendered MD.

Reply:
I don't think pre-processing the tables is a good solution, especially as we want the contributing file to be rendered correctly on both GitHub and our mdbook docs.

Maybe down the line we could experiment with another format; for instance, I believe GitHub also supports AsciiDocs, I imagine mdbook can too. However that's way out of scope here.

I'll take a look at the formatting next time I'm on a big enough screen, but I'm sure it's fine 🫠

Reply:
Just had a look on a wider screen. Looks fine other than two lines:

<details><summary>Details</summary>
<p>

![image](https://github.com/user-attachments/assets/5a3b1244-2b77-474e-af40-8437058b2eb0)

</p>
</details> 

[Line 67](https://github.com/DataHearth/nixvim/blob/2650c1852840bf3df4fca271c9750b25711a06a5/CONTRIBUTING.md?plain=1#L67) (`extraOption`) and [Line 73](https://github.com/DataHearth/nixvim/blob/2650c1852840bf3df4fca271c9750b25711a06a5/CONTRIBUTING.md?plain=1#L73) (`installPackage`) seem to have an extra space somewhere